Consistant renderer crash on windows from v0.33.9+

I get a consistent crash on windows 10 64bit (tested on 2 different machines).
It happen in my app when I close a tab and free a bunch of objects, possibly linked with others.

Please note that:
- There is no crash report sent, even if crash-reporter is setup in both main and renderer sides
- It does not happen on mac nor linux
- I get the crash on all versions from 0.33.9 to 0.34.1 but not on 0.33.8 and before.

#### How to reproduce
- Download [one of the distributions of my app](https://tweakstyle.com/download.html#all-downloads)
- Run the app.asar with a newer version of electron (v0.33.9 or greater)
- After the welcome screen, double click on index.html from the sample project
- In the newly created pane, open one of the files file by clicking on it
- Close the tab => Renderer Crash

I made more tests. I get the crash only with the x64 build of electron. The ia32 build is working fine.

Here is a quick capture of how to reproduce http://recordit.co/iIEOjH6nb7
Note: I open the html file with a double click

I'm trying to build electron to get the full stack, but I'm stuck with the following error at build step:
```
ninja: Entering directory `out\D'
[360/589] CXX obj\vendor\node\src\node.string_bytes.obj
c:\users\felicien\source\repos\electron\vendor\node\src\string_bytes.cc(191) : warning C4003: not enough actual parameters for macro 'V'
[377/589] CXX obj\atom\browser\ui\electron_lib.file_dialog_win.obj
FAILED: ninja -t msvc -e environment.x64 -- "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\x86_amd64\cl.exe" /nologo /showIncludes /FC @obj\atom\browser\ui\electron_lib.file_dialog_win.obj.rsp /c ..\..\atom\browser\ui\file_dialog_win.cc /Foobj\atom\browser\ui\electron_lib.file_dialog_win.obj /Fdobj\electron_lib.cc.pdb
c:\users\felicien\source\repos\electron\atom\browser\ui\file_dialog_win.cc(7) : fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory
[377/589] CXX obj\atom\browser\electron_lib.browser_win.obj
FAILED: ninja -t msvc -e environment.x64 -- "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\x86_amd64\cl.exe" /nologo /showIncludes /FC @obj\atom\browser\electron_lib.browser_win.obj.rsp /c ..\..\atom\browser\browser_win.cc /Foobj\atom\browser\electron_lib.browser_win.obj /Fdobj\electron_lib.cc.pdb
c:\users\felicien\source\repos\electron\atom\browser\browser_win.cc(7) : fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory
[377/589] CXX obj\atom\browser\electron_lib.web_view_manager.obj
ninja: build stopped: subcommand failed.
```

Not sure if it's a common error. I have visual studio 2013 Community

--- a/atom/common/crash_reporter/crash_reporter_win.cc
+++ b/atom/common/crash_reporter/crash_reporter_win.cc
@@ -80,7 +80,7 @@ struct ExceptionHandlerRecord {
   unsigned char thunk[12];
 };
 
-void RegisterNonABICompliantCodeRange(void* start, size_t size_in_bytes) {
+bool RegisterNonABICompliantCodeRange(void* start, size_t size_in_bytes) {
   ExceptionHandlerRecord* record =
       reinterpret_cast<ExceptionHandlerRecord*>(start);
 
@@ -117,17 +117,17 @@ void RegisterNonABICompliantCodeRange(void* start, size_t size_in_bytes) {
 
   // Protect reserved page against modifications.
   DWORD old_protect;
-  CHECK(VirtualProtect(
-      start, sizeof(ExceptionHandlerRecord), PAGE_EXECUTE_READ, &old_protect));
-  CHECK(RtlAddFunctionTable(
-      &record->runtime_function, 1, reinterpret_cast<DWORD64>(start)));
+  return VirtualProtect(start, sizeof(ExceptionHandlerRecord),
+                        PAGE_EXECUTE_READ, &old_protect) &&
+         RtlAddFunctionTable(&record->runtime_function, 1,
+                             reinterpret_cast<DWORD64>(start));
 }
 
 void UnregisterNonABICompliantCodeRange(void* start) {
   ExceptionHandlerRecord* record =
       reinterpret_cast<ExceptionHandlerRecord*>(start);
 
-  CHECK(RtlDeleteFunctionTable(&record->runtime_function));
+  RtlDeleteFunctionTable(&record->runtime_function);
 }
 #endif  // _WIN64
 
@@ -184,6 +184,7 @@ void CrashReporterWin::InitBreakpad(const std::string& product_name,
     LOG(ERROR) << "Cannot initialize out-of-process crash handler";
 
 #ifdef _WIN64
+  bool registered = false;
   // Hook up V8 to breakpad.
   {
     // gin::Debug::SetCodeRangeCreatedCallback only runs the callback when
@@ -192,9 +193,10 @@ void CrashReporterWin::InitBreakpad(const std::string& product_name,
     size_t size = 0;
     v8::Isolate::GetCurrent()->GetCodeRange(&code_range, &size);
     if (code_range && size)
-      RegisterNonABICompliantCodeRange(code_range, size);
+      registered = RegisterNonABICompliantCodeRange(code_range, size);
   }
-  gin::Debug::SetCodeRangeDeletedCallback(UnregisterNonABICompliantCodeRange);
+  if (registered)
+    gin::Debug::SetCodeRangeDeletedCallback(UnregisterNonABICompliantCodeRange);
 #endif
 }
 

